packages/nimble-components/src/rich-text/editor/testing/rich-text-editor.pageobject.ts

```diff
@@ -92,6 +92,8 @@ export class RichTextEditorPageObject {
 
     public async clickFooterButton(button: ToolbarButton): Promise<void> {
         const toggleButton = this.getFormattingButton(button);
+        const event = new Event('mousedown', { bubbles: true });
+        toggleButton!.dispatchEvent(event);
         toggleButton!.click();
```

Comment:
Do you know why the click behavior needs to fire the mousedown event?
From what I can tell the template listens for click events:
https://github.com/ni/nimble/blob/f159a084527c7e53bac99dd43182c10c2f2f628f/packages/nimble-components/src/rich-text/editor/template.ts#L52
and invoking the click method does emit those click events: https://jsbin.com/bohelazuwu/edit?html,console,output

It's not clear to me why this change is needed

Reply:
It is specific to the focus behavior of the buttons in the toolbar. The _clicking buttons should update the checked state of the toggle button with focus_ collection of tests fail without it because they assert that the `tab-index` of each button is correct after clicking on it, but the fast toolbar updates the `tab-index` based on `mousedown` due to the ordering of mouse, click, and focus events.

Reply:
Ahh I see the discussion here: https://github.com/microsoft/fast/pull/6836

And a real interactive click goes through the mousedown, mouseup, click, progression. Same for a touch-based click, it should fire the same mouse events I think.

Maybe at some point we should have some helpers used in the page object that simulate real clicks better. Not something we need in this PR.

Could you sanity check that the focus behavior is working  / no weird issues on mobile? I vaguely remember issues with the rich text buttons on ios safari, I wonder if those behaviors are impacted @jattasNI

Reply:
I don't recall the issues we saw in the past. I tried on iOS with storybook builds from main and from this PR and didn't notice any behavior differences. (Note if others want to try, #1639 makes it hard but I got it to work in landscape mode). The experience kinda sucks because clicking the buttons gives keyboard focus to the text field which causes the keyboard to show and overlap the buttons. But if you hide the keyboard the buttons show the states that I'd expect.

Reply:
I verified that it works correctly on Windows desktop browsers (Chrome, Edge, and Firefox) and that it works correctly on iOS safari. There is a small lag on iOS safari with the buttons being pressed and the styling updating, but that doesn't seem to be related to this change because it is also there on the version of storybook published to nimble.ni.dev.

Reply:
Yea we should do a pass on our quick interaction things (like increment / decrement) and se touch-action: manipulation for a better mobile experience: https://developer.mozilla.org/en-US/docs/Web/CSS/touch-action#manipulation

Can you make an issue for that?